cmd/settlement/paypal.go

```diff
@@ -118,69 +109,67 @@ func PaypalEmailTemplate(inPath string, outPath string) (err error) {
 }
 
 var (
-	paypalSettlementCmd = &cobra.Command{
+	// PaypalSettlementCmd is the paypal command
+	PaypalSettlementCmd = &cobra.Command{
 		Use:   "paypal",
 		Short: "provides paypal settlement",
 	}
 
-	emailPaypalSettlementCmd = &cobra.Command{
+	// EmailPaypalSettlementCmd provides population of a templated email
+	EmailPaypalSettlementCmd = &cobra.Command{
 		Use:   "email",
 		Short: "provides population of a templated email",
-		Run: func(cmd *cobra.Command, args []string) {
-			if err := PaypalEmailTemplate(input, out); err != nil {
-				log.Printf("failed to perform email templating: %s\n", err)
-				os.Exit(1)
-			}
-		},
+		Run:   cmd.Perform("email", EmailPaypalSettlement),
 	}
 
-	completePaypalSettlementCmd = &cobra.Command{
+	// CompletePaypalSettlementCmd provides completion of paypal settlement
+	CompletePaypalSettlementCmd = &cobra.Command{
 		Use:   "complete",
 		Short: "provides completion of paypal settlement",
-		Run: func(cmd *cobra.Command, args []string) {
-			if err := PaypalCompleteSettlement(input, out, txnID); err != nil {
-				log.Printf("failed to perform complete: %s\n", err)
-				os.Exit(1)
-			}
-		},
+		Run:   cmd.Perform("complete", CompletePaypalSettlement),
 	}
 
-	transformPaypalSettlementCmd = &cobra.Command{
+	// TransformPaypalSettlementCmd provides transform of paypal settlement for mass pay
+	TransformPaypalSettlementCmd = &cobra.Command{
 		Use:   "transform",
 		Short: "provides transform of paypal settlement for mass pay",
-		Run: func(cmd *cobra.Command, args []string) {
-			transformFailed := func(err error) {
-				log.Printf("failed to perform transform: %s\n", err)
-				os.Exit(1)
-			}
-
-			payouts, err := settlement.ReadFiles(strings.Split(input, ","))
-			if err != nil {
-				transformFailed(err)
-			}
-
-			if err := PaypalTransformForMassPay(
-				cmd.Context(),
-				payouts,
-				currency,
-				decimal.NewFromFloat(rate),
-				out,
-			); err != nil {
-				transformFailed(err)
-			}
-		},
+		Run:   cmd.Perform("transform", TransformPaypalSettlement),
 	}
 )
 
+// EmailPaypalSettlement create the email to send to the
+func EmailPaypalSettlement(cmd *cobra.Command, args []string) error {
+	return PaypalEmailTemplate(viper.GetString("input"), viper.GetString("out"))
+}
+
+// TransformPaypalSettlement transforms a paypal settlement
+func TransformPaypalSettlement(cmd *cobra.Command, args []string) error {
+	payouts, err := settlement.ReadFiles(strings.Split(viper.GetString("input"), ","))
+	if err != nil {
+		return err
+	}
+
+	return PaypalTransformForMassPay(
+		cmd.Context(),
+		payouts,
+		viper.GetString("currency"),
+		decimal.NewFromFloat(viper.GetFloat64("rate")),
+		viper.GetString("out"),
+	)
+}
+
+// CompletePaypalSettlement added complete paypal settlement
+func CompletePaypalSettlement(cmd *cobra.Command, args []string) error {
+	return PaypalCompleteSettlement(
```

Comment:
Since we have this level of indirection should we move more of the logic into the `settlement` module instead of keeping it here in `cmd`? Thoughts @maikelmclauflin @husobee?

Reply:
yeah, we can certainly do that: #528